src/lib/middleware/origin-middleware.ts

```diff
@@ -0,0 +1,27 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { IUnleashConfig } from '../types';
+
+export const originMiddleware = ({
+    getLogger,
+    flagResolver,
+}: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>) => {
+    const logger = getLogger('/middleware/origin-middleware.ts');
+    logger.debug('Enabling origin middleware');
+
+    return (req: Request, _: Response, next: NextFunction) => {
+        if (!flagResolver.isEnabled('originMiddleware')) {
+            return next();
+        }
+
+        const isUI = !req.headers.authorization;
```

Comment:
Is there something else we'd check? I assume we're already authorized at this point, so probably this is enough

Reply:
Yep, I also think it's enough, and I believe the placement means this only runs after authorization.

Reply:
> I may be missing some edge cases

If there's any I think this will help us figure that out (somehow)